### PR TITLE
Collapse repeated elisions (#42 ported to the ES6 codebase; closes #41)

### LINKED ISSUE #41: Collapse repeated elision into one line
Currently, when diffing arrays, elements that are objects and equivalent on both sides get elided with a `...`.  However, the fact that the elision marker is there is important to highlight the position of the differences in the array.

In some cases (especially when diffing API responses and similar), the arrays might be very long, and the output gets rather tedious. Instead of one elision marker for each elided object, I suggest to replace repetated elision with a single line.

E.g.,
```diff
% json-diff <(echo '{ "hello": [{},{},{},{},{},{},1] }') <(echo '{ "hello": [{},{},{},{},{},{},2] }')
 {
   hello: [
     ...
     ...
     ...
     ...
     ...
     ...
-    1
+    2
   ]
 }
```
would instead become
```diff
% json-diff <(echo '{ "hello": [{},{},{},{},{},{},1] }') <(echo '{ "hello": [{},{},{},{},{},{},2] }')
 {
   hello: [
     ... (6 entries)
-    1
+    2
   ]
 }
```

For smaller arrays, it would get noisy and reduce clarity to use the elisions everywhere.  Therefore, I think it makes sense to by default only collapse e.g. 5 or more elisions into one line. Probably makes sense to provide a command-line flag to set this threshold, though!

### PR DESCRIPTION
**Pull request by FireyFly**
As per [feedback in #42][1], here's a new PR against the new codebase post moving to ES6. It also incorporates the suggestion of letting the user decide the threshold at which we collapse the repeated `...`'s.

I opted to default to no collapsing. However, the `--max-elisions` intentionally does nothing (not even warn) in 'full' mode; this is intentional so that a user can set a shell alias to `json-diff --max-elisions 5` or similar to customize the behaviour.

The `--help` output gets a bit wonkily aligned now due to the length of `--max-elisions COUNT`. At first I tried to call the argument just `N` instead, but unfortunately this fails with dreamopt [due to this regex][2]. Ah well.

[1]: https://github.com/andreyvit/json-diff/pull/42#issuecomment-998448291
[2]: https://github.com/andreyvit/dreamopt.js/blob/master/lib/dreamopt.coffee#L12

### CODE REVIEW

**Comment by ewoudenberg** [comment]
Thank you! There was one typo in the unit tests, which I fixed.

**Comment by FireyFly** [comment]
oh whoops.. yeah, wasn't sure what to call the option initially. Thanks!

File: lib/colorize.js [hunk 1]
```diff
@@ -8,27 +8,37 @@ const Theme = {
...
 }
 
-const subcolorizeToCallback = function (key, diff, output, color, indent) {
+const subcolorizeToCallback = function (options, key, diff, output, color, indent) {
   let subvalue
   const prefix = key ? `${key}: ` : ''
   const subindent = indent + '  '
 
+  const outputElisions = (n) => {
+    if (n < (options.maxElisions ?? Infinity)) {
```
Comment (ljharb) [inline lib/colorize.js:17]: this syntax is a breaking change; see #79.

Comment (FireyFly) [reply lib/colorize.js:17]: oops! thanks for the heads-up, and for the quick fix @ewoudenberg. I've been travelling recently so missed this comment; sorry about the breakage.

**Pull request merged**